modules/book/addbook.rb

```diff
@@ -0,0 +1,40 @@
+module AddBook
+  def add_book
+    publish_date = input_date
+    publisher = input_publisher
+    cover_state = input_cover_state
+    book = Book.new(cover_state, publisher, publish_date)
+    @books << book
+    label = add_label(book)
+    book.add_label(label)
```

Comment:
- Good job done so far, after implementing the custom setter method, to avoid errors in your code, please change this line to:

```ruby
   book.label = label
```